### ISSUE TITLE
Specify cloud provider during DB creation

### ISSUE THREAD

**Issue by gstachni** [datastax/astra-cli#53]
This feedback was captured using a feedback form on datastax.com

While trying to automate an astra db creation via astra CLI I noticed that there's no way to specify the cloud provider. I reviewed the documentation, as well as the help command and it doesn't seem to be a way to change the default cloud provider. Would be great to have an option to do that.

**Comment by clun** [comment]
Al righty

### 1/ A couple of new commands

For genericity and could help the user as well

```
astra db list-clouds
astra streaming list-clouds
```

<img width="141" alt="Screenshot 2023-03-09 at 15 09 13" src="https://user-images.githubusercontent.com/726536/224181159-75cf1671-9861-4756-9cca-893d1a269574.png">

```
astra db list-regions-serverless --cloud aws
```
<img width="542" alt="Screenshot 2023-03-09 at 15 09 44" src="https://user-images.githubusercontent.com/726536/224181208-20fc5d6b-83b1-4203-9bbe-23b5e9a3d30e.png">


### 2/ Implementation of cloud check

#### If you provide invalid cloud

```
db create invalid --cloud cumulonimbus --region us-east1
```

> ```
> [ERROR] Invalid Cloud Provider value for 'cumulonimbus', valid options are aws, gcp and azure (InvalidCloudProviderException)
> [ERROR] Try 'astra help <your_command>' to get help
> [INFO]  Use astra db list-clouds or astra streaming list-clouds to list available cloud providers.
> ```


#### if you provide valid cloud with invalid region for that cloud

```
db create invalid --cloud aws --region us-east1
```

<img width="950" alt="Screenshot 2023-03-09 at 15 22 20" src="https://user-images.githubusercontent.com/726536/224182952-e8592130-4c79-4aa5-b0a4-02db5ddfbc26.png">

#### if you do not provide any cloud...back to default behaviour

```
db create correct  --region us-east-1
```
<img width="670" alt="Screenshot 2023-03-09 at 15 33 27" src="https://user-images.githubusercontent.com/726536/224184372-0dd5641c-b3cb-4f15-8ec9-e30c31d87e9d.png">

**Comment by synedra** [comment]
Shouldnt the free providers be flagged as such?  Sent from my iPhoneOn Apr 6, 2023, at 12:55 PM, Cedrick Lunven ***@***.***> wrote:﻿
Closed #53 as completed.

**Comment by clun** [comment]
I will create a dedicated ticket for this one.
